src/components/ProjectSearch/TextSearch.js

```diff
@@ -0,0 +1,73 @@
+import { Component, DOM as dom, createFactory } from "react";
+import classnames from "classnames";
+
+import Svg from "../shared/Svg";
+import _ManagedTree from "../shared/ManagedTree";
+const ManagedTree = createFactory(_ManagedTree);
+
+import "./TextSearch.css";
+
+export default class TextSearch extends Component {
+  constructor(props: Props) {
+    super(props);
+  }
+
+  renderFile(file, expanded) {
+    return dom.div(
+      {
+        className: "file-result",
+        style: {}
+      },
+      Svg("arrow", {
+        className: classnames({
+          expanded
+        })
+      }),
+      file.filepath
+    );
+  }
+
+  renderLine(match) {
+    return dom.div(
+      {
+        className: "result"
+      },
+      dom.span(
+        {
+          className: "line-number"
+        },
+        match.line
+      ),
+      dom.span(
+        { className: "line-match", style: { display: "flex", grow: 1 } },
+        match.value
+      )
+    );
+  }
+
+  renderResults() {
+    const { results } = this.props;
+    return ManagedTree({
```

Comment:
this is very complex, it feels like most of these callbacks belong to ManagedTree. Maybe a job for a future refactoring